samples/boards/stm32/pm_lptim/src/main.c

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright (c) 2016 Intel Corporation.
+ * Copyright (c) 2020 STMicroelectronics.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+
+#include <string.h>
+#include <soc.h>
+#include <device.h>
+#include <drivers/gpio.h>
+#include <sys/printk.h>
+
+struct device *gpio_bn, *gpio_led;
+
+bool ready_to_sleep;
+
+/* in msec */
+/* setting to K_TICKS_FOREVER willl activate the deepsleep mode */
+#define SLEEP_TIME 3000
+
+#define DEMO_DESCRIPTION	\
+	"Demo Description\n"	\
+	"Application is demonstrating the low power" \
+	"with the LPTIM used as systick.\n" \
+	"The system goes to sleep mode on k_sleep() when pressing the button\n" \
+	"(led off) and wakes up by the lptim timeout expiration (led on)\n\n" \
+
+/* The devicetree node identifier for the "sw0" alias. */
+#define SW0_NODE DT_ALIAS(sw0)
+/* GPIO for the User push button */
+#if DT_NODE_HAS_STATUS(SW0_NODE, okay)
+#define SW	DT_GPIO_LABEL(SW0_NODE, gpios)
+#define PIN	DT_GPIO_PIN(SW0_NODE, gpios)
+#if DT_PHA_HAS_CELL(SW0_NODE, gpios, flags)
+#define SW0_FLAGS DT_GPIO_FLAGS(SW0_NODE, gpios)
+#else
+#define SW0_FLAGS 0
+#endif
+#else
+/* A build error here means your board isn't set up to input a button. */
+#error "Unsupported board: sw0 devicetree alias is not defined"
+#endif
+
+/* The devicetree node identifier for the "led0" alias. */
+#define LED0_NODE DT_ALIAS(led0)
+/* GPIO for the Led */
+#if DT_NODE_HAS_STATUS(LED0_NODE, okay)
+#define LED0	DT_GPIO_LABEL(LED0_NODE, gpios)
+#define PIN0	DT_GPIO_PIN(LED0_NODE, gpios)
+#if DT_PHA_HAS_CELL(LED0_NODE, gpios, flags)
+#define LED0_FLAGS DT_GPIO_FLAGS(LED0_NODE, gpios)
+#else
+#define LED0_FLAGS 0
+#endif
+#else
+/* A build error here means your board isn't set up to blink an LED. */
+#error "Unsupported board: led0 devicetree alias is not defined"
+#endif
+
+static struct gpio_callback gpio_cb;
+
+void button_pressed(struct device *dev, struct gpio_callback *cb,
+		    uint32_t pins)
+{
+	ARG_UNUSED(dev);
+	ARG_UNUSED(pins);
+	ARG_UNUSED(cb);
+
+	/* Set pin to HIGH/LOW */
```

Comment:
The comment is confusing.  How about:

    /* Turn LED off */

if this needs a comment at all.